synapse/tests/test_lib_net.py

```diff
@@ -2,8 +2,198 @@
 
 from synapse.tests.common import *
 
+logger = logging.getLogger(__name__)
+
 class NetTest(SynTest):
 
+    def test_lib_net_chan_txfini(self):
+        class FiniableLink(s_net.Link):
+            def tx(msg, **kwargs):
```

Comment:
is ``self`` missing from this definition?

Reply:
The first arg passed to the fn is the mesg. Looked weird to me as well

Reply:
```
diff --git a/synapse/tests/test_lib_net.py b/synapse/tests/test_lib_net.py
index c44a2f6..434b203 100644
--- a/synapse/tests/test_lib_net.py
+++ b/synapse/tests/test_lib_net.py
@@ -8,12 +8,12 @@ class NetTest(SynTest):
 
     def test_lib_net_chan_txfini(self):
         class FiniableLink(s_net.Link):
-            def tx(msg, **kwargs):
+            def tx(self, msg):
                 logger.error(msg)
 
         msg = '%r' % (('cool', {}),)
         with self.getLoggerStream('synapse.tests.test_lib_net', msg) as stream:
-            chan = s_net.Chan(FiniableLink, None)
+            chan = s_net.Chan(FiniableLink(), None)
             chan.txfini(data=('cool', {}))
             stream.seek(0)
             self.isin(msg, stream.read())
``` seems to work

Reply:
this was wrong, fixed